app/views/HomePage/TopSellingItem/index.tsx

```diff
@@ -0,0 +1,109 @@
+                categories {
+                    name
+                }
+                numberOfPages
+            }
+        }
+    }
+`;
+
+function TopSellingItem() {
+    const {
+        data: topSellingItemsResponse,
+    } = useQuery<TopSellingItemQuery>(
+        TOP_SELLING_ITEMS,
+    );
+
+    const [visibleItems, setVisibleItems] = useState(3);
+    const debounceTimeoutRef = React.useRef<number>();
+
+    React.useEffect(() => {
+        const onNumberOfVisibleItemsChange = () => {
+            window.clearTimeout(debounceTimeoutRef.current);
+            debounceTimeoutRef.current = window.setTimeout(() => {
+                const width = document.documentElement.clientWidth;
+                if (width < 710) {
+                    setVisibleItems(1);
+                } else if (width < 1200) {
+                    setVisibleItems(2);
+                } else {
+                    setVisibleItems(3);
+                }
+            }, 200);
+        };
+
+        window.addEventListener('resize', onNumberOfVisibleItemsChange);
+        onNumberOfVisibleItemsChange();
+
+        return () => {
+            window.removeEventListener('resize', onNumberOfVisibleItemsChange);
+        };
+    }, []);
+
+    return (
+        <Carousel
+            className={styles.carousel}
+            numberOfVisibleItems={visibleItems}
+        >
+            <div className={styles.carouselWrapper}>
+                <CarouselButton action="prev" />
+                <div className={styles.itemList}>
+                    {topSellingItemsResponse?.books?.results?.map((books, i) => (
+                        <CarouselItem
+                            className={styles.carouselItem}
+                            order={i + 1}
+                        >
+                            <Link
+                                to={{
+                                    pathname: routes.bookList.path,
+                                    state: { book: books.title },
+                                }}
+                                className={styles.gradeItem}
+                                linkElementClassName={styles.link}
+                            >
+                                <img
+                                    className={styles.preview}
+                                    src={books.image?.url ?? undefined}
+                                    alt={books.title}
+                                />
+                                <div className={styles.description}>
+                                    {books.title}
+                                </div>
+                                <div className={styles.price}>
+                                    {`Rs. ${books.price}`}
+                                </div>
+                            </Link>
+                        </CarouselItem>
+                    ))}
+                </div>
+                <CarouselButton className={styles.carouselButton} action="next" />
```

Comment:
there is no `carouselButton` styling definition. add it in the `styles.css` or remove it.

Reply:
Done

Reply:
It won't work passing key value in this case as ordering of books depends on the position of array item